src/harness/fourslashImpl.ts

```diff
@@ -822,6 +822,27 @@ export class TestState {
         });
     }
 
+    public baselineInlayHints(span: ts.TextSpan = { start: 0, length: this.activeFile.content.length }, preferences?: ts.UserPreferences): void{
+        interface HasPosition { position: number; }
+        const sortHints = (a: HasPosition, b: HasPosition) => {
+            return a.position - b.position;
+        };
+
+        const baselineFile = this.getBaselineFileNameForContainingTestFile();
+        const fileName = this.activeFile.fileName;
+        const hints = this.languageService.provideInlayHints(fileName, span, preferences);
+        const annotations = ts.map(hints.sort(sortHints), hint => {
+            const span = { start: hint.position, length: hint.text.length };
+            const startLc = this.languageServiceAdapterHost.positionToLineAndCharacter(fileName, span.start);

```

Comment:
you only use `startLc` for the line, so I would just extract the line out immediately

Reply:
I use the character for creating the underline.

Reply:
Duh!